src/meta/src/stream/sink.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::anyhow;
+use risingwave_connector::sink::catalog::SinkCatalog;
+use risingwave_connector::sink::{SinkConfig, SinkImpl};
+use risingwave_pb::catalog::Sink as ProstSinkCatalog;
+
+use crate::{MetaError, MetaResult};
+
+pub async fn validate_sink(
```

Comment:
Shall we pass the `SinkType` by the way for validation?

Reply:
The `SinkType` may be passed all along to the connector node.

Reply:
If you mean `SinkType` in `SinkCatalog`, it is actually an enum representing the sink's append-only attribute (`AppendOnly`, `Upsert`, etc.). We do have another `sink_type` in our code, but only exists within `RemoteSink` and actually conveys the connector type info (jdbc, iceberg, etc.). I know this is really confusing and I've renamed it to `connector_type` in our rust code. Since this `connector_type` only exists within `RemoteSink`, we'd better figure this out in `RemoteSink`'s method instead of here in meta.

Reply:
It's actually included in `SinkCatalog`, so the connector node is able to get the sink type info.

Reply:
> Shall we pass the SinkType by the way for validation? 

IIUC, we validate availability and schema compatibility for rw and downstream systems. `SinkType` should be deprecated as we no longer see data in sink as rw internal.

Reply:
The connector node will not connect to the meta, so we'd better include the `SinkType` in, for example, `SinkConfig`, so that in sink validation and sink initialization we can check the sink type if necessary.

Reply:
BTW, I just checked the code of remote sink. Seems that we added a generic const `APPEND_ONLY` in the `RemoteSink`, but we didn't use this const at all. I think we can remove this generic const and instead store the sink type as a new field in the `RemoteSink` and when we build the rpc request we can put the sink type in the request.